frontend/src/components/AddUpdate/AddActivityUpdate.tsx

```diff
@@ -0,0 +1,81 @@
+import {useNavigate} from "react-router-dom";
+import {FormEvent, useState} from "react";
+import {TripUpdateActivityEmissionsDto} from "../../model/updateDtos/TripUpdateActivityEmissionsDto";
+
+type AddActivityUpdateProps = {
+    tripId: string
+    updateActivityEmissions: (id: string, tripUpdateActivityEmissionsDto: TripUpdateActivityEmissionsDto) => void
+}
+
+export default function AddActivityUpdate({updateActivityEmissions, tripId}: AddActivityUpdateProps) {
+    const navigate = useNavigate()
+    const [amountOfGolfRounds, setAmountOfGolfRounds] = useState<number>(0)
+    const [amountOfSkiingDays, setAmountOfSkiingDays] = useState<number>(0)
+    const [amountOfBeautyDays, setAmountOfBeautyDays] = useState<number>(0)
+    const [customActivityItem, setCustomActivityItem] = useState<string>(``)
+    const [customActivityItemEmission, setCustomActivityItemEmission] = useState<number>(0)
+    const [amountOfCustomActivityItem, setAmountOfCustomActivityItem] = useState<number>(0)
+
+    const onUpdate = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const tripUpdateActivityEmissionsDto: TripUpdateActivityEmissionsDto = {
+            amountOfGolfRounds: amountOfGolfRounds,
+            amountOfSkiingDays: amountOfSkiingDays,
+            amountOfBeautyDays: amountOfBeautyDays,
+            customActivityItem: customActivityItem,
+            customActivityItemEmission: customActivityItemEmission,
+            amountOfCustomActivityItem: amountOfCustomActivityItem,
+        }
+
+        updateActivityEmissions(tripId, tripUpdateActivityEmissionsDto);
+        setAmountOfGolfRounds(0)
+        setAmountOfSkiingDays (0)
+        setAmountOfBeautyDays(0)
+        setCustomActivityItem (``)
+        setCustomActivityItemEmission(0)
+        setAmountOfCustomActivityItem(0)
```

Comment:
What is this block for - sry - I do not understand this.

Reply:
THis will be the topic of my 1on1 today. Without these setters the update function ist not working